winch/codegen/src/regset.rs

```diff
@@ -62,35 +99,48 @@ impl RegSet {
     /// register class to determine the bitset that requires updating.
     pub fn free(&mut self, reg: Reg) {
         let index = reg.hw_enc() as u32;
```

Comment:
Along with the above comment, let's assert that `index` is in range here rather than silently truncating (the `as u32` cast) and then left-shifting off the end (`1 << index`) -- perhaps `assert!(reg.hw_enc() < MAX_REGS);` and define `MAX_REGS` near where we would use `u32` (or `u64` as per other comment) for the bitfield above.